### PR TITLE
1619 - Fix language switching from UserSettingScreen

### LINKED ISSUE #1619: [Quest] - Fix change language from user settings screen
**Describe the bug**
Fix change language from user settings screen

**To Reproduce**
1. Go to user setting screen by clicking `profile` from the side-menu (drawer)
2. Select a different language from the language drop down
3. Observe language change not effected

**Expected behavior**
Selecting a language should reload the screen with new language. 

**Possible causes** 
Activity/Fragment not getting refreshed. The activity should be recreated for new language to take effect.

### PR DESCRIPTION
**Pull request by qaziabubakar-vd**
Update the UserSettingViewModel.kt to follow event pattern as quest app

**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #1619

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

**Review by pld — DISMISSED** [review]
lgtm! can you bump up the coverage then merge pls

**Review by pld — DISMISSED** [review]

File: android/engine/src/main/java/org/smartregister/fhircore/engine/ui/usersetting/UserSettingScreen.kt [hunk 1]
```diff
@@ -62,12 +64,15 @@ import org.smartregister.fhircore.engine.ui.theme.BlueTextColor
 import org.smartregister.fhircore.engine.ui.theme.DividerColor
 import org.smartregister.fhircore.engine.ui.theme.LighterBlue
 
+const val USER_SETTING_ROW_TEST_TAG = "userSettingRowTestTag"
+
 @Composable
 fun UserSettingScreen(
   modifier: Modifier = Modifier,
-  userSettingViewModel: UserSettingViewModel = hiltViewModel()
+  userSettingViewModel: UserSettingViewModel = hiltViewModel(),
```
Comment (ellykits) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/ui/usersetting/UserSettingScreen.kt:72]: How about we provide the `onEvent` method as a lambda parameter to this function including the rest of the items (properties/functions) need from the `UserSettingViewModel` to be able to test this better? We will not need to provide the `UserSettingViewModel` instance anymore and we can instantiate it on the fragment instead.

**Review by ellykits — COMMENTED** [review]
Code looks great after the refactor. Left a suggestion.

**Comment by qaziabubakar-vd** [comment]
I've made the requested changes, and updated the failing tests as well.

**Review by pld — APPROVED** [review]

**Review by ellykits — APPROVED** [review]

**Pull request merged**